domains/client/domain-executor/src/fraud_proof.rs

```diff
@@ -152,7 +152,13 @@ where
             // `finalize_block` execution proof.
             let pre_state_root = as_h256(&local_receipt.trace[local_trace_index as usize - 1])?;
             let post_state_root = as_h256(local_root)?;
-            let execution_phase = ExecutionPhase::FinalizeBlock;
+            let extrinsics = self.block_body(block_hash)?;
+            let execution_phase = ExecutionPhase::FinalizeBlock {
+                total_extrinsics: extrinsics
+                    .len()
+                    .try_into()
+                    .expect("extrinsic_index must fit into u32"),
```

Comment:
Perhaps `local_trace_index - 1` is simpler.

Reply:
Sorry for the ambiguity, I mean `total_extrinsics` are serve as the trace index (just like `idx` in `ExecutionPhase::ApplyExtrinsic(idx)`), perhaps `ExecutionPhase::FinalizeBlock(local_trace_index - 1)` is simpler.

Reply:
Simpler but less understandable honestly. It's straightforward to think of `u32` as the extrinsic index in `ApplyExtrinsic(u32)` because each extrinsic inherently has an index and this also appears in other places (e.g., `Phase` in frame-system), but not the case when it comes to what `u32` means for `FinalzieBlock`, I made it explicit intentionally.